src/m365/entra/commands/group/group-user-remove.ts

```diff
@@ -0,0 +1,222 @@
+import { entraUser } from '../../../../utils/entraUser.js';
+import { GraphBatchRequest, GraphBatchRequestResponse } from '../../../../utils/types.js';
+import { validation } from '../../../../utils/validation.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  groupDisplayName?: string;
+  ids?: string;
+  userNames?: string;
+  role?: string;
+  suppressNotFound?: boolean;
+  force?: boolean;
+}
+
+class EntraGroupUserRemoveCommand extends GraphCommand {
+  private readonly roleValues = ['Owner', 'Member'];
+
+  public get name(): string {
+    return commands.GROUP_USER_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes users from a Microsoft Entra ID group';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        ids: typeof args.options.ids !== 'undefined',
+        userNames: typeof args.options.userNames !== 'undefined',
+        role: typeof args.options.role !== 'undefined',
+        suppressNotFound: !!args.options.suppressNotFound,
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --groupId [groupId]'
+      },
+      {
+        option: '-n, --groupDisplayName [groupDisplayName]'
+      },
+      {
+        option: '--ids [ids]'
+      },
+      {
+        option: '--userNames [userNames]'
+      },
+      {
+        option: '-r, --role [role]',
+        autocomplete: this.roleValues
+      },
+      {
+        option: '--suppressNotFound'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId !== undefined && !validation.isValidGuid(args.options.groupId)) {
+          return `'${args.options.groupId}' is not a valid GUID for option 'groupId'.`;
+        }
+
+        if (args.options.ids !== undefined) {
+          const ids = args.options.ids.split(',').map(i => i.trim());
+          if (!validation.isValidGuidArray(ids)) {
+            const invalidGuid = ids.find(id => !validation.isValidGuid(id));
+            return `'${invalidGuid}' is not a valid GUID for option 'ids'.`;
+          }
+        }
+
+        if (args.options.userNames !== undefined) {
+          const isValidUserPrincipalNameArray = validation.isValidUserPrincipalNameArray(args.options.userNames.split(',').map(u => u.trim()));
```

Comment:
Same goes here: let's introduce a new version of the method, which returns all invalid values at once, rather than just the first one.